feat(header): add language selector (#253)
eclipse-mnestix/mnestix-browser: wiki/Adding-Languages.md

## Adding your Language to Mnestix

### Available Languages

Currently, Mnestix Browser supports three languages, available from the language selector located on the top-right side:
- English (default language)

> - Deustch (German)

Reviewer: ```suggestion
- Deutsch (German)
```

> - Deustch (German)

Reviewer 2: ```suggestion
- Deutsch (German)
```

> - Español (Spanish)

Reviewer: I'm a bit confused here: How do we support Spanish if there is no translation file and no routing for it currently?

Reviewer 3: It's another PR, just merged it today

> ⚠️ **Note:** In versions previous to adding language selector, language can be switched by changing the language code part of
> the URL as in the following example:
> ```markdown
>       https://mnestix.com/en/list --> (English selected by having 'en')
>       https://mnestix.com/de/list --> (Deustch selected by having 'de')
> ```

> > This functonality persists after the addition of language selector.

Reviewer 3: Same, I think we can remove this sentence

### How do I add my own language to Mnestix?

Mnestix browser supports the addition of more languages locally.

To add a language, changes are needed on:
- ``` src/locale``` folder → Add JSON file with translations.
- ``` src/i18n/routing.ts``` file → Add entry with language code for translations.

#### Add a JSON file with the translations

Messages around the browser are loaded from language files named ```{language_code}.json``` where language code represents 
the two letter code for the language (i.e. English → en).

To add a language create a file with these naming convention under ``` src/locale``` folder and copy the contents of another language
file inside the same folder. Then, change the strings to your language for all the messages as in the example below:

- In English version ````en.json````
```yaml
    "buttons": {
              "add": "Add",
              "back": "Back",
              "cancel": "Cancel",
              "create": "Create Rule",
              "delete": "Delete",
              "edit": "Edit",
              "save": "Save"
            }
```

> - In Deustch version ````de.json````

Reviewer 2: ```suggestion
- In german version ````de.json````
```

```yaml
    "buttons": {
              "add": "Hinzufügen",
              "back": "Zurück",
              "cancel": "Abbrechen",
              "create": "Regel erstellen",
              "delete": "Löschen",
              "edit": "Bearbeiten",
              "save": "Speichern"
            }
```
#### Add language code referencing the translations

For the language to be selectable from the Mnestix browser, the language router needs to know it exists, this can be indicated in
``` src/i18n/routing.ts``` file.

Open the file and add your language to ````locales```` as in example below:

````tsx
locales: ['en', 'de'] --> Add your language code (i.e. 'es') --> locales: ['en', 'de', 'es']
````

> > ⚠️ **Important:** After making the changes, you might need to build again the Mnestix browser for the changes to take effect.

Reviewer 2: ```suggestion
> ⚠️ **Important:** After making the changes, you might need to rebuild the Mnestix browser for the changes to take effect.
```

### Contributing your language

> Adding a language is a time-consuming task. If you have invested the time to create a new language support, we would appreciate if you

Reviewer 2: ```suggestion
Adding a language is a time-consuming task. If you have invested the time to create a new language support, we would appreciate it if you
```
